### PR TITLE
[android] Follow-up on the new project structure

### PR DESCRIPTION
**Pull request by rtsisyk**

### CODE REVIEW

File: android/app/src/flavors/gms-enabled/AndroidManifest.xml [hunk 1]
```diff
@@ -1,7 +0,0 @@
-<?xml version="1.0" encoding="utf-8"?>
```
Comment (rtsisyk) [inline android/app/src/flavors/gms-enabled/AndroidManifest.xml:1]: Please note that this file wasn't compiled in at all, even before all changes with the project structure. I tested that it isn't really needed and the app works without it on the oldest API=21 supported.

File: android/app/src/main/assets/00_NotoNaskhArabic-Regular.ttf [hunk 2]
```diff
@@ -0,0 +1 @@
+../../../../../data/00_NotoNaskhArabic-Regular.ttf
```
Comment (rtsisyk) [inline android/app/src/main/assets/00_NotoNaskhArabic-Regular.ttf:1]: I moved android/app/src/assets to android/app/src/main/assets to match the default Android Studio layout. This folder is picked up by gradle automatically without any hacks with assets.srcDirs = ["assets"].

File: android/app/src/web/java/app/organicmaps/location [hunk 3]
```diff
@@ -0,0 +1 @@
+../../../../google/java/app/organicmaps/location
```
Comment (rtsisyk) [inline android/app/src/web/java/app/organicmaps/location:1]: This was the easiest way to support gms and non-gms builds without creating additional dimension in build.gradle. I don't to create complexity like "packageGoogleGmsWorldDebug" and  "packageFDroidNonGmsNonWorldDebug". Symlinks allowed me to keep "packageGoogleDebug" and  "packageFdroidDebug".

File: android/app/src/nofirebase/java/app/organicmaps/util/CrashlyticsUtils.java
Comment (rtsisyk) [inline android/app/src/nofirebase/java/app/organicmaps/util/CrashlyticsUtils.java:1]: Raised another PR to remove this file completely https://github.com/organicmaps/organicmaps/pull/5937

**Comment by rtsisyk** [comment]
The new flavors structure now matches [recommendations](https://developer.android.com/build/build-variants#sourceset-build):

image

```
ls -1 android/app/src
androidTest
fdroid
firebase
google
huawei
main
nofirebase
test
web
```

**Comment by rtsisyk** [comment]
All `srcDirs +=` hacks from build.gradle besides "firebase"/"nonfirebase" has been removed. The later is addressed by #5937.

**Comment by rtsisyk** [comment]
Please read carefully https://stackoverflow.com/questions/37439120/product-flavor-duplicate-class-found
There are just three options:

1. Manipulations with srcDirs as it was before:

```
    if (googleMobileServicesEnabled) {
       java.srcDirs += 'src/flavors/gms-enabled'
     } else {
       java.srcDirs += 'src/flavors/gms-disabled'
     }
```

2. Symlinks - the current option

gms-enabled implementation is in src/google/java/app/organicmaps/location, non-gms enabled  src/fdroid/java/app/organicmaps/location. Huawei and Web are symlinked to `src/google/java/app/organicmaps/location`

3. More dimensions in the matrix

./gradlew -Parm64 packageGoogleGmsWorldAutoDebug
./gradlew -Parm64 packageFDroidNonGmsNoWorldAutoDebug

This is recommended ways without any hacks, but "packageGoogleGmsWorldAutoDebug" looks crazy...

I think that (2) is a good compromise solution.

**Review by vng — APPROVED** [review]
LGTM, but I'm not the right guy to review gradles :)

**Pull request merged**